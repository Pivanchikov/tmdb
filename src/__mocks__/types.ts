import { Movie } from '../state/types'

export type Genres = {
  id: number
  name: string
}

export type GenresSelectOption = {
  value: number | string
  label: string
}

export type TransformMovies = {
  [key: string]: Movie
}
