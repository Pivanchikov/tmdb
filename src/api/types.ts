export type MovieGetParams = {
  genre: string
  country: string
  year: number | string
  languageCode: string
  includeAdults?: boolean
}
