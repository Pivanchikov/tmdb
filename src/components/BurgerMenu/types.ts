export type MenuProps = {
  items: {
    top: string[]
    middle: string[]
    bottom: string[]
  }
}
