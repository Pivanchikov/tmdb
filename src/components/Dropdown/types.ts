export type DropdownProps = {
  options: { value: string | number; label: string }[]
  onChange?: (e: React.SyntheticEvent<EventTarget>) => void
  emptyValue?: { value: string | number; label: string }
  defaultValue?: string
  testId?: string
}
