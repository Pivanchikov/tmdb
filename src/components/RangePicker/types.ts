export type RangePickerProps = {
  step: number
  min: number
  max: number
  values: number[]
  label: React.ReactNode
  onChange: (years: number[]) => void
}
