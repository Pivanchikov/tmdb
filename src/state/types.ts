export type MoviesState = {
  items: []
  byId: Record<string, never>
  genres: []
  selectedMovie: null | Movie
  isLoading: boolean
  isFoundMovie: boolean
  popularMovies: Movie[]
  trendingMovies: Movie[]
  nowPlayingMovies: Movie[]
  upcomingMovies: Movie[]
  topRatedMovies: Movie[]
}

export type Movie = {
  adult: boolean
  backdrop_path: string
  genre_ids: number[]
  id: number
  original_language: string
  original_title: string
  overview: string | null
  popularity: number
  poster_path: string | null
  release_date: string
  title: string
  video: boolean
  vote_average: number
  vote_count: number
}

export type fetchSingleMovieParams = { genre: string; country: string; year: number | string }
