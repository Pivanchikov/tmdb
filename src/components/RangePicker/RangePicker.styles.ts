export const styles = theme => ({
  trackContainer: {
    height: 36,
    display: 'flex',
    width: '100%',
  },
  track: {
    height: 5,
    width: '100%',
    borderRadius: 4,
    alignSelf: 'center',
  },
  thumb: {
    height: 25,
    width: 25,
    borderRadius: 4,
    backgroundColor: theme.palette.secondary.main,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    boxShadow: `0 2px 6px ${theme.palette.secondary.dark}`,
  },
  thumbSquare: {
    height: 10,
    width: 5,
  },
  label: {
    color: theme.palette.primary.main,
    // eslint-disable-next-line @typescript-eslint/prefer-as-const
    textAlign: 'center' as 'center',
  },
})
