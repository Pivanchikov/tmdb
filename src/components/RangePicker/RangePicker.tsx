import { Range, getTrackBackground } from 'react-range'
import { makeStyles, Typography } from '@material-ui/core'

import { styles } from './RangePicker.styles'
import { primaryMain, secondaryDark } from '../../theme/theme'
import { RangePickerProps } from './types'
import { ClassNameMap } from '@material-ui/styles'

const useStyles = makeStyles(styles)

type renderTrackProps = {
  props: {
    style: React.CSSProperties
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    ref: React.RefObject<any>
    onMouseDown: (e: React.MouseEvent) => void
    onTouchStart: (e: React.TouchEvent) => void
  }
  children: React.ReactNode
  classes: ClassNameMap
  values: number[]
  min: number
  max: number
}

type renderThumbProps = {
  props: {
    key: number
    style: React.CSSProperties
    tabIndex?: number
    'aria-valuemax': number
    'aria-valuemin': number
    'aria-valuenow': number
    draggable: boolean
    role: string
    onKeyDown: (e: React.KeyboardEvent) => void
    onKeyUp: (e: React.KeyboardEvent) => void
  }
  isDragged: boolean
  classes: ClassNameMap
}

const renderTrack = ({ props, children, classes, values, min, max }: renderTrackProps) => {
  const rangeStyles = {
    background: getTrackBackground({
      values,
      colors: [secondaryDark, primaryMain],
      min,
      max,
    }),
  }

  return (
    <div className={classes.trackContainer}>
      <div ref={props.ref} style={rangeStyles} className={classes.track}>
        {children}
      </div>
    </div>
  )
}

const renderThumb = ({ props, isDragged, classes }: renderThumbProps) => (
  <div {...props} className={classes.thumb}>
    <div
      style={{
        backgroundColor: isDragged ? primaryMain : secondaryDark,
      }}
      className={classes.thumbSquare}
    />
  </div>
)

export const RangePicker = ({ step, min, max, values, onChange, label }: RangePickerProps) => {
  const classes = useStyles()

  return (
    <>
      <Range
        values={values}
        step={step}
        min={min}
        max={max}
        onChange={onChange}
        renderTrack={({ props, children }) => renderTrack({ props, children, classes, values, min, max })}
        renderThumb={({ props, isDragged }) => renderThumb({ props, isDragged, classes })}
      />
      <Typography className={classes.label}>{label}</Typography>
    </>
  )
}
