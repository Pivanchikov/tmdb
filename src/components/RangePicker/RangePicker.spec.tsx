import { render as testingLibraryRender } from '@testing-library/react'

import { RangePicker } from './RangePicker'
import { RangePickerProps } from './types'

describe('movie dropdown', () => {
  const requiredProps: RangePickerProps = {
    step: 1,
    min: 1,
    max: 20,
    label: 'production year',
    onChange: jest.fn(),
    // eslint-disable-next-line no-magic-numbers
    values: [1],
  }
  const render = () => testingLibraryRender(<RangePicker {...requiredProps} />)

  it('matches snapshot', () => {
    const { asFragment } = render()
    expect(asFragment()).toMatchSnapshot()
  })
})
