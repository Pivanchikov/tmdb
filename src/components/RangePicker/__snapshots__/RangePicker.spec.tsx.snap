// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie dropdown matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-trackContainer-1"
  >
    <div
      class="makeStyles-track-2"
    >
      <div
        aria-valuemax="20"
        aria-valuemin="1"
        aria-valuenow="1"
        class="makeStyles-thumb-3"
        draggable="false"
        role="slider"
        style="position: absolute; z-index: 0; cursor: grab; user-select: none; transform: translate(NaNpx, NaNpx);"
        tabindex="0"
      >
        <div
          class="makeStyles-thumbSquare-4"
          style="background-color: rgb(204, 204, 204);"
        />
      </div>
    </div>
  </div>
  <p
    class="MuiTypography-root makeStyles-label-5 MuiTypography-body1"
  >
    production year
  </p>
</DocumentFragment>
`;
